src/Mvc/Mvc.ViewFeatures/test/HtmlHelperComponentExtensionsTests.cs

```diff
@@ -310,6 +330,26 @@ protected override async Task OnParametersSetAsync()
             }
         }
 
+        private class OnAfterRenderComponent : ComponentBase
+        {
+            [Parameter] public State State { get; set; }
```

Comment:
This is pretty obscure. Why is it a `[Parameter]` and why is it called `State`?

I'd expect something more like:

```
bool didInvokeOnAfterRender;
```

Reply:
Oh wait I see why you did it as a `[Parameter]`

Maybe the value could still be called something more obvious, e.g.

```
        private class OnAfterRenderState
        {
            public bool DidInvokeOnAfterRender { get; set; }
        }
```

Reply:
Updated